lighthouse-core/report/html/renderer/dom.js

```diff
@@ -216,13 +217,18 @@ class DOM {
    * @template {string} T
    * @param {T} query
    * @param {ParentNode} context
+   * @return {ParseSelector<T>}
    */
   find(query, context) {
     const result = context.querySelector(query);
     if (result === null) {
       throw new Error(`query ${query} not found`);
     }
-    return result;
+
+    // Because we control the report layout and templates, use the simpler
+    // `typed-query-selector` types that don't require differentiating between
+    // e.g. HTMLAnchorElement and SVGAElement. See https://github.com/GoogleChrome/lighthouse/issues/12011.
+    return /** @type {ParseSelector<T>} */ (result);
```

Comment:
I'd love to not cast here, but the types aren't quite compatible, testing is totally broken if you try to filter out SVG elements at runtime (https://github.com/jsdom/jsdom/issues/2734), and if you try to do naive filtering of the type tsc will give you things like `HTMLElementTagNameMap[keyof HTMLElementTagNameMap] & HTMLAnchorElement`, which, when parameterized, gives you leftovers like `SVGAElement & HTMLAnchorElement`, which is _not_ the empty set of properties so isn't eliminating anything from the union of types :)